client/src/app/routes/nova/SlotPage.tsx

```diff
@@ -31,34 +33,40 @@ export default function SlotPage({
     const { slotCommitment: slotCommitmentDetails, slotCommitmentId } = useSlotDetails(network, slotIndex);
     const { slotManaBurned } = useSlotManaBurned(slotIndex);
     const [slotStats] = useSlotStats(slotIndex);
+    const [formatManaAmounts, setFormatManaAmounts] = useState(false);
 
     const parsedSlotIndex = parseSlotIndexFromParams(slotIndex);
     const slotStatus = getSlotStatusFromLatestSlotCommitments(parsedSlotIndex, latestSlotCommitments);
     const slotFromSlotCommitments = latestSlotCommitments.find((slot) => slot.slotCommitment.slot === parsedSlotIndex);
     const slotTimeRange = parsedSlotIndex && slotIndexToUnixTimeRange ? slotIndexToUnixTimeRange(parsedSlotIndex) : null;
     const slotTimestamp = getSlotTimestamp(slotTimeRange);
 
-    const dataRows: IPageDataRow[] = [
+    const rmc =
+        slotFromSlotCommitments?.slotCommitment?.referenceManaCost?.toString() ??
+        slotCommitmentDetails?.referenceManaCost?.toString() ??
+        "-";
+    const manaBurned = slotManaBurned?.manaBurned ? formatAmount(slotManaBurned?.manaBurned, manaInfo, formatManaAmounts) : "-";
+    const copyManaBurned = slotManaBurned?.manaBurned ? formatAmount(slotManaBurned?.manaBurned, manaInfo, true) : "-";
+    const slotData: CardInfoProps[] = [
         {
-            label: "Slot Index",
+            title: "Slot Index",
             value: parsedSlotIndex ?? "-",
         },
+        { title: "Timestamp", value: slotTimestamp ?? "-" },
         {
-            label: "Commitment Id",
-            value: slotCommitmentId ?? "-",
+            title: "RMC",
+            value: formatAmount(rmc, manaInfo, formatManaAmounts),
+            onClickValue: () => setFormatManaAmounts(!formatManaAmounts),
+            copyValue: formatAmount(rmc, manaInfo, formatManaAmounts, true),
```

Comment:
once this is ready, imo we can merge, looks amazing already 🚀

Reply:
it could be its own isue and PR if this escalates, but yes ideally we should unify the amount copy feature, we should review the whole usage and see where it is still needed to be improved